### DISCUSSION TITLE
GDScript parser/tokenizer errors appear at the output/terminal, which is annoying

### DISCUSSION THREAD

**Discussion by Xrayez** [godotengine/godot-proposals#4084] [Scripting]
I think parse/tokenizer errors should not be printed to the console (in production builds). For example, when I type `for`, I always get the following output:

![image](https://user-images.githubusercontent.com/17108460/155200605-d74ea4b8-570e-49b8-840e-74d713a18bac.png)

I have code complete delay set to low value as well which amplifies this issue:

![image](https://user-images.githubusercontent.com/17108460/155201408-a5cdef6c-324c-4fa5-8cb3-e6a3d2acf932.png)

I think those prints should only be enabled in `DEV_ENABLED` builds.

I'm not sure whether this issue is present in Godot 4.x, but this is currently an issue in Godot 3.x, which would be nice to fix.

**Comment by azur-wolve** [comment]
Lol every time that look at some "low-level" engine thread, its rare that you @Xrayez are not involved hehe.

It looks like they leave it there to enforce users to help with bug-solving, I guess.
Can't disable it on the settings?

**Reply by Xrayez** [reply]
I think there's no way to disable this, other than recompiling the engine. If there's a way to disable this, then it would probably mean disabling *all* print errors, but the point is that seeing errors is actually beneficial during development. Yet as a user, I'm not interested in developing the engine (I'm both engine developer and user, though!). Those errors come from implementation detail which normal users cannot even have access to.
